x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommand.java

```diff
@@ -864,7 +864,7 @@ private boolean askCertSigningRequest(Terminal terminal) {
             terminal.println("IT IS IMPORTANT THAT YOU REMEMBER THIS PASSWORD AND KEEP IT SECURE");
             terminal.println("");
             final char[] password = readPassword(terminal, "CA password: ", true);
-            return new CertificateTool.CAInfo(caCert, keyPair.getPrivate(), true, password.length == 0 ? null : password);
+            return new CertificateTool.CAInfo(caCert, keyPair.getPrivate(), true, password);
```

Comment:
This change is the fix. But is there any reason why this was intentionally set to `null` for empty char array? I wonder whether the change would introduce any subtle side effect. @tvernum

Reply:
Inspecting the uses of this returned `CAInfo` instance, I am pretty sure this change is not impactful.

However, looking at [this](https://github.com/elastic/elasticsearch/blob/e2fc95d97a21f01a39da001476ca50b9caf9521b/x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertificateTool.java#L387) other use of `CAInfo`, there seems to be a pattern where a `null` value is used to signal that a password prompt is further required. I believe the original code here intended for this same behavior.